File: backend/src/main/java/com/seniorcare/api/model/HealthData.java
package com.seniorcare.api.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "health_data")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class HealthData {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @Column(name = "heart_rate")
    private Integer heartRate;
    
    @Column(name = "steps_count")
    private Integer stepsCount;
    
    @Column(name = "calories_burned")
    private Double caloriesBurned;
    
    @Column(name = "sleep_duration_minutes")
    private Integer sleepDurationMinutes;
    
    @Column(name = "sleep_quality")
    private String sleepQuality;
    
    @Column(name = "blood_pressure_systolic")
    private Integer bloodPressureSystolic;
    
    @Column(name = "blood_pressure_diastolic")
    private Integer bloodPressureDiastolic;
    
    @Column(name = "blood_oxygen")
    private Double bloodOxygen;
    
    @Column(name = "body_temperature")
    private Double bodyTemperature;
    
    @Column(name = "recorded_at", nullable = false)
    private LocalDateTime recordedAt;
    
    @Column(name = "device_id")
    private String deviceId;
    
    @Column(name = "data_source")
    private String dataSource;
}

File: watch-app/src/main/java/com/seniorcare/watch/domain/model/HealthData.kt
package com.seniorcare.watch.domain.model

import java.util.Date

/**
 * 건강 데이터 모델
 */
data class HealthData(
    val id: Long? = null,
    val heartRate: Int? = null,
    val stepsCount: Int? = null,
    val caloriesBurned: Double? = null,
    val sleepDurationMinutes: Int? = null,
    val sleepQuality: String? = null,
    val bloodPressureSystolic: Int? = null,
    val bloodPressureDiastolic: Int? = null,
    val bloodOxygen: Double? = null,
    val bodyTemperature: Double? = null,
    val recordedAt: Date = Date(),
    val deviceId: String? = null,
    val dataSource: DataSource = DataSource.WATCH
) {
    enum class DataSource {
        WATCH,
        PHONE,
        MANUAL,
        EXTERNAL_DEVICE
    }
}

/**
 * 건강 데이터 요약 모델
 */
data class HealthSummary(
    val date: Date,
    val averageHeartRate: Int? = null,
    val minHeartRate: Int? = null,
    val maxHeartRate: Int? = null,
    val totalSteps: Int? = null,
    val totalCaloriesBurned: Double? = null,
    val totalSleepMinutes: Int? = null,
    val averageBloodPressureSystolic: Int? = null,
    val averageBloodPressureDiastolic: Int? = null,
    val averageBloodOxygen: Double? = null
)
